### DISCUSSION TITLE
Introducing `yewtube` - a fork of mps-youtube with no Youtube API Key required

### DISCUSSION THREAD

**Discussion by iamtalhaasghar** [mps-youtube/yewtube#1202] [General]
Because this project have not been actively maintained lately, I have forked it and implemented some exciting features such as:
1. No Youtube API key required to run `yewtube`
2. `yewtube` can play audio in vlc with no gui.
3. `yewtube` inherits all original features of `mps-youtube`
4. `yewtube` uses `yt-dlp` instead of `youtube-dl` as the main focus of `yt-dlp` is to add new features and patches while also keeping up to date with the original `youtube-dl` project.
4. `yewtube` is available on `pypi.org`

Installation instructions can be found [here](https://iamtalhaasghar.github.io/yewtube/#installation).
`yewtube` is actively maintained so i request you all to try it out and report your issues / feature requests [here](https://github.com/iamtalhaasghar/yewtube/issues). Pull requests are also welcomed.

**Comment by rachmadaniHaryono** [comment]
- do you track any issue that is fixed by your fork?
- what is the difference between your fork and https://github.com/otisdog8/termitube? 
	- currently i favor your fork because you open the issue section
- any plan to merge it upstream?

**Comment by iamtalhaasghar** [comment]
> * do you track any issue that is fixed by your fork?

Yes,

>     * what is the difference between your fork and https://github.com/otisdog8/termitube?

The goal for `yewtube` is that I want it to be a fully featured Youtube client so that you don't have to use Youtube in a browser. 

>       * currently i favor your fork because you open the issue section

Thank you!

>     * any plan to merge it upstream?

If the maintainers want then sure why not.

**Comment by rachmadaniHaryono** [comment]
any reason why there is readme.rst and readme.md?

**Comment by iamtalhaasghar** [comment]
> any reason why there is readme.rst and readme.md?

I am using `README.md` and updating it to reflect latest information about my fork `yewtube`. I think `README.rst` is being used by maintainers of `mps-youtube` for similar purpose.

**Comment by rachmadaniHaryono** [comment]
i think your pypi page not working because you don't set `long_description_content_type` to `text/markdown`

related https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/#including-your-readme-in-your-package-s-metadata

**Comment by iamtalhaasghar** [comment]
Please raise issues [here](https://github.com/iamtalhaasghar/yewtube/issues)

**Comment by mlliarm** [comment]
Hey, thanks for this.

Nice to see that all my playlists from `mps-youtube` were ported in `yewtube` without having to do anything.

Also my the issue I had (#1166) is fixed now, so thanks. I'll go and let that poor fella know.

**Comment by Elias-Gill** [comment]
Really appreciate this. Thank you for bringing us back this gorgeous app

**Comment by lassedong** [comment]
WOW. I have been waiting for something like this, for a long time.

But I have one issue, so far.
When using yewtube download-command, like "d 6" I do not get any list of download options.
Instead yewtube will immediately start to download the selected video as webm.,

Please help.

**Comment by iamtalhaasghar** [comment]
@lassedong checkout https://github.com/iamtalhaasghar/yewtube/issues/50

Edit: this is a known issue, you can track progress using the above link.

**Comment by lassedong** [comment]
> @lassedong checkout [iamtalhaasghar/yewtube#50](https://github.com/iamtalhaasghar/yewtube/issues/50)

Yes, It is yewtube, version 2.8.2

I am running Linux Mint 20.3 Cinnamon.
I installed yewtube with sudo pip install yewtube, and it installed and replaced mps-youtube.
So if I type mpsyt, it will start yewtube.

**Comment by mlliarm** [comment]
@lassedong I'm running yewtube of `v.2.8.0` on an Lubuntu 18.04.6 and I just type `yt` to fire up yewtube.

~~@iamtalhaasghar has the command needed to start `yewtube` changed in newer than `v.2.8.0` versions?~~

Just upgraded to `v.2.8.2`. I just type `yt` to fire up yewtube. Can you try this, @lassedong?

**Comment by lassedong** [comment]
> @lassedong I'm running yewtube of `v.2.8.0` on an Lubuntu 18.04.6 and I just type `yt` to fire up yewtube.
> 
> ~@iamtalhaasghar has the command needed to start `yewtube` changed in newer than `v.2.8.0` versions?~
> 
> Just upgraded to `v.2.8.2`. I just type `yt` to fire up yewtube. Can you try this, @lassedong?

It is the same result.

I can start yewtube, by typing "yt" also.
But the problem is the same.

**Comment by lassedong** [comment]
I have now tried to download and installl yewtube on  my Windows 11 machine.

First I installed phyton 3.7.0. Then I installed pipx, as instructed, and finally I installed yewtube, also as instructed.

After invoking the yt-command, yewtube will start, but I still don't get any choice to select the quality of a video to downlod.
Yewtube immediately starts to download the video.

![image](https://user-images.githubusercontent.com/102852294/161952372-ac38fd17-2f2e-4f84-b897-e57f9caf51cb.png)

**Comment by iamtalhaasghar** [comment]
@lassedong read my comment again. I have edited it

**Comment by lassedong** [comment]
> 

Okay, Thanks.

I will follow.

Are you also aware, that download of playlists does not work?

**Comment by iamtalhaasghar** [comment]
> > 
> 
> Okay, Thanks.
> 
> I will follow.
> 
> Are you also aware, that download of playlists does not work?

yes,

**Comment by lassedong** [comment]
Hi.... Is this project dead or still being developed?

**Comment by ids1024** [comment]
It's nice to see a fork of mps-youtube that's gotten rid of reliance on the YouTube data API v3 (I got involved with mps-youtube mainly to fix its reliance on the v2 of the data API, which was being disabled.) And added other improvements.

@iamtalhaasghar Since no on who was involved in development in the past seems interested in maintaining mps-youtube and pafy (including me), would you be interested in being added as an owner to the mps-youtube GitHub org? And on PyPI.

**Comment by iamtalhaasghar** [comment]
> It's nice to see a fork of mps-youtube that's gotten rid of reliance on the YouTube data API v3 (I got involved with mps-youtube mainly to fix its reliance on the v2 of the data API, which was being disabled.) And added other improvements.
> 
> @iamtalhaasghar Since no on who was involved in development in the past seems interested in maintaining mps-youtube and pafy (including me), would you be interested in being added as an owner to the mps-youtube GitHub org? And on PyPI.

Sure why not,

**Comment by ids1024** [comment]
Invitations sent on Github and PyPI.

**Reply by iamtalhaasghar** [reply]
Can I rename `mps-youtube` to `yewtube`? @ids1024 

P.S: i've contacted github support team to reattach my fork to `mps-youtube`. Will push my changes upstream very soon.

**Reply by ids1024** [reply]
Sure. That's perhaps a better name anyway.

If you want to merge the changes from `yewtube` into this repository, you should be able to transfer issues from that repository, but you'd have to transfer it to the Github org first. So I guess you could rename that to `yewtube-old` or something, transfer it, transfer the issues, archive `yewtube-old` with a note about the other repository and rename the `mps-youtube` repository and organization. I *think* things should redirect cleanly.

**Reply by Sophira** [reply]
I would be very careful with using a name that has the exact same pronunciation as "YouTube", even if for no other reason than it makes it difficult to talk about it in speech only.

**Reply by iamtalhaasghar** [reply]
Ok then, you guys can suggest some cool names and we can choose one with the highest number of likes / votes.

**Comment by mlliarm** [comment]
Yewtube sounds good to me.

---
Michail L. Liarmakopoulos, MSc
https://member.acm.org/~liarmakopoulosm

**Comment by pechkaeltsin** [comment]
Thank you for the wonderful YouTube CLI.
Could you please tell me how to open the video not via m8u3 download, but via url (`https://www.youtube.com/watch?v=blablabla`)? Just, for example, iina plays m8u3 without picture, only sound. 
Thanks again, I'm just glad to be able to use YouTube without their horrible site.
